airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -511,15 +523,17 @@ def __init__(self, *args, **kwargs):
             start_date=self.start_date,
         )
 
-    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs):
+    def path(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs: Any) -> str:
+        assert isinstance(stream_slice, dict), "stream_slice must be a dictionary"
```

Comment:
I find this a bit dangerous as we could want to change this to any class that support the interface `__get__` and this would start failing. Is there are reason why we need to validate that it is an instance of a dict?